src/error.rs

```diff
@@ -1,40 +1,34 @@
-use base_x;
-use std::{error, fmt};
+/// Type alias to use this library's [`Error`] type in a `Result`.
+pub type Result<T> = std::result::Result<T, Error>;
 
 /// Error types
 #[derive(PartialEq, Eq, Clone, Debug)]
 pub enum Error {
-    UnknownBase,
+    /// Unknown base code.
+    UnknownBase(char),
+    /// Invalid string.
     InvalidBaseString,
 }
 
-pub type Result<T> = ::std::result::Result<T, Error>;
-
-impl fmt::Display for Error {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        f.write_str(error::Error::description(self))
-    }
-}
-
-impl error::Error for Error {
-    fn description(&self) -> &str {
-        use Error::*;
-
-        match *self {
-            UnknownBase => "Unknown base",
-            InvalidBaseString => "Invalid base string",
+impl std::fmt::Display for Error {
```

Comment:
Please keep the original import `use std::{error, fmt}` instead of using the fully qualified name.